src/PHPStanStaticTypeMapper/TypeMapper/UnionTypeMapper.php

```diff
@@ -215,24 +204,6 @@ private function matchPhpParserUnionType(UnionType $unionType, string $typeKind)
         return $this->resolveTypeWithNullablePHPParserUnionType(new PhpParserUnionType($phpParserUnionedTypes));
     }
 
-    /**
-     * @param TypeKind::* $typeKind
-     */
-    private function resolveAllowedStandaloneTypeInUnionType(
-        Type $unionedType,
-        string $typeKind
-    ): Identifier|Name|null|PHPParserNodeIntersectionType|ComplexType {
-        if ($unionedType instanceof NullType) {
-            return new Identifier('null');
-        }
-
-        if ($unionedType instanceof ConstantBooleanType && ! $unionedType->getValue()) {
-            return new Identifier('false');
-        }
-
-        return $this->phpStanStaticTypeMapper->mapToPhpParserNode($unionedType, $typeKind);
```

Comment:
This created a union with `false` even in PHP 8.0 and 8.1, which is a bug.
Instead I've moved the logic to bool/null mapper itself, where it belongs. Added new union type enum to verify where the type belongs.

Reply:
false in union in php 8.0 seems valid https://3v4l.org/Y5D5H5#v8.0.30

Reply:
The one that not allowed in php 8.0 is nullable false 

https://3v4l.org/iWbuF

Reply:
I'll try to get false|null too :+1: